Add load balancer, and quick start documentation. (#3)
microservices-march/platform: README.md

# Microservices March Demo Architecture Platform (WIP)
Shared infrastructure for the Microservices March demo architecture.

> ## Usage

Reviewer: ```suggestion
## Overview

```

> The aim is to set things up as manually as possible without involving something heavy like Kubernetes. Future iterations might have examples of more advanced setups.

Reviewer: ```suggestion
The aim of this demo architecture is to set up a microservices architecture as manually as possible without involving something heavy like Kubernetes. Future iterations might have examples of more advanced setups.
```

> ## Running just shared infrastructure
> If you are running some subset of the services manually, just run `docker-compose up` from this repository to start up the message queue.  This does not set up the NGINX load balancer.

Reviewer: ```suggestion
```
You will already cover the manual setup in [L15](https://github.com/microservices-march/platform/pull/3/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R15)

Author: This one is a bit different but maybe I've worded it poorly.  There's three things you can do:
1. Manual setup (prebuilt containers everywhere, started with mostly `docker run` commands and some `docker-compose up` commands).
2. Quick setup (one big docker compose file)
3. Running a subset of services locally on your machine (how most devs work in my experience).  So for example I might be running `messenger` with just node on my mac, and need to have the rabbit queue available for that.  

How does this alternative wording sound?

> If you are working on one of the services, such as [messenger](https://github.com/microservices-march/messenger), that depends on a piece of shared infrastructure to function, you can run `docker-compose up` from this repository to provide that infrastructure.  This does not set up the NGINX load balancer.

Reviewer: Your suggestion should work! The original wording was indeed unclear/missing some info.

> ## Setting up the whole demo architecture.

Reviewer: ```suggestion
## Setting Up the Demo Architecture

```

Starting up the entire system including the applications is not part of the role of this repository. However, instructions are included here for simplicity.

We provide two methods to set things up:

> 1. [Quick Setup](docs/quick-setup.md): Based on `docker-compose`. Use this one if you just want to get started quickly.
> 1. [Manual Setup](docs/manual-setup.md): Uses some `docker-compose` but mostly raw docker commands. Use this if you want go through what the quick start does for you automatically for learning.

Reviewer: ```suggestion

1. [Manual Setup](https://github.com/microservices-march/platform/blob/main/docs/manual-setup.md): Follow this guide. if you want to manually go through most steps. Recommended for learning. 
2. [Quick Setup](https://github.com/microservices-march/platform/blob/main/docs/quick-setup.md): Follow this guide if you want to get started quickly. Automates mosts steps covered in the Manual Setup guide.

```
Let's put the manual setup first to try to help guide users towards it. There is no need to specify that we are using Docker Compose to various extents. The respective guide descriptions do that already.

Author: Good suggestion, thanks!

> ## NGINX (Load Balancer)

Reviewer: ```suggestion
## Components
### NGINX (Load Balancer)

```
It feels a bit weird to simply add NGINX and RabbitMQ at the end of the doc. If we group both under a Components header (or something similar) it will help make the structure of the doc clearer/more logical.

NGINX is used in front of the entire archtecture to provide load balancing to all the services.
Currently only the `messenger` service accepts HTTP requests, so it simply routes to that.

> ## RabbitMQ (Message Queue)

Reviewer: ```suggestion
### RabbitMQ (Message Queue)
```
